features/ui/select/select.tsx

```diff
@@ -0,0 +1,256 @@
+      strokeLinecap="round"
+      strokeLinejoin="round"
+    />
+  </DownCaret>
+);
+
+const Placeholder = ({ children, ...props }: PlaceholderProps) => (
+  <components.Placeholder {...props}>
+    {props.selectProps.iconSrc && (
+      <Icon src={props.selectProps.iconSrc} alt="" />
+    )}
+    {children}
+  </components.Placeholder>
+);
+
+const customStyles: StylesConfig = {
+  control: (provided, state) => ({
+    ...provided,
+    maxWidth: "20rem",
+    borderWidth: "1px",
+    borderColor: state.selectProps.error
+      ? `${color("error", 300)({ theme })}`
+      : state.isFocused
+      ? `${color("primary", 300)({ theme })}`
+      : `${color("gray", 300)({ theme })}`,
+    borderRadius: "8px",
+    boxShadow: state.isFocused
+      ? state.selectProps.error
+        ? "0px 1px 2px rgba(16, 24, 40, 0.05), 0px 0px 0px 4px #FEE4E2"
+        : "0px 1px 2px rgba(16, 24, 40, 0.05), 0px 0px 0px 4px #F4EBFF;"
+      : "0px 1px 2px rgba(16, 24, 40, 0.05)",
+    padding: "0.625rem 0.8125rem",
+    margin: "6px 0",
+    backgroundColor: state.isDisabled
+      ? `${color("gray", 50)({ theme })}`
+      : "#FFFFFF",
+
+    "&:hover": {
+      borderColor: state.selectProps.error
+        ? `${color("error", 300)({ theme })}`
+        : state.isFocused
+        ? `${color("primary", 300)({ theme })}`
+        : `${color("gray", 300)({ theme })}`,
+      cursor: "pointer",
+    },
+  }),
+
+  menu: (provided) => ({
+    ...provided,
+    maxWidth: "20rem",
+    boxShadow:
+      "0px 12px 16px -4px rgba(16, 24, 40, 0.1), 0px 4px 6px -2px rgba(16, 24, 40, 0.05)",
+    borderRadius: "8px",
+    margin: "0.5rem 0",
+  }),
+
+  menuList: (provided) => ({
+    ...provided,
+    maxHeight: "20rem",
+  }),
+
+  option: (provided, state) => ({
+    ...provided,
+    color: `${color("gray", 900)({ theme })}`,
+    lineHeight: "1.5rem",
+    padding: "0.625rem 0.875rem",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "space-between",
+    backgroundColor:
+      state.isFocused || state.isSelected
+        ? `${color("primary", 25)({ theme })}`
+        : "#FFFFFF",
+
+    "&:hover": {
+      backgroundColor: `${color("primary", 25)({ theme })}`,
+    },
+  }),
+
+  valueContainer: (provided) => ({
+    ...provided,
+    margin: "0",
+    padding: "0",
+  }),
+
+  placeholder: (provided) => ({
+    ...provided,
+    color: `${color("gray", 500)({ theme })}`,
+    lineHeight: "1.5rem",
+    display: "flex",
+    alignContent: "center",
+    margin: "0",
+  }),
+
+  singleValue: (provided) => ({
+    ...provided,
+    lineHeight: "1.5rem",
+    display: "flex",
+    alignContent: "center",
+    margin: "0",
+  }),
+};
+
+export function SelectComponent({
+  label,
+  hintMsg,
+  errorMsg,
+  error,
+  iconSrc,
+  ...Props
+}: Props) {
+  return (
+    <div>
+      {label && <Label id="reactSelectId">{label}</Label>}
+      <Select
+        {...Props}
+        iconSrc={iconSrc}
+        error={Props.isDisabled ? false : error}
```

Comment:
Nitpick: For consistency I'd rename this prop to `disabled` as you did in the other input components. This way you also isolate the usage of `react-select` from the rest of the app.

Reply:
I've had a go implementing this, but I'm not sure if I've done it the right way. By adding my own `disabled` prop we keep consistent, even though it uses `react-select` `isDisabled` prop under the hood? (See PR #21 ). Would this isolate the usage of `react-select` the way you suggest? I could see how you could swap out for another package and link our disabled prop to whatever the package's disabled functionality was without impacting other code using our `disabled` prop.